src/transformers/trainer.py

```diff
@@ -1325,6 +1325,10 @@ def _wrap_model(self, model, training=True, dataloader=None):
         if not training:
             return model
 
+        # add torchrun support for XLA, but currently DDP doesn't support XLA yet
+        if is_torch_tpu_available():
+            return model
+
```

Comment:
No the model needs to be wrapped in a `DistributedDataParallel` for TPU,

Reply:
Thanks. Limited the case to torchrun and clarify that currently DDP doesn't work with torch.distributed XLA backend yet (https://github.com/pytorch/pytorch/issues/79164)

Reply:
In this case we should wait until it's supported. This will break the current TPU support.